src/test/shared/filesystemUtilities.test.ts

```diff
@@ -101,33 +101,17 @@ describe('filesystemUtilities', () => {
         })
     })
 
-    describe('isContainedWithinDirectory', () => {
+    it('isInDirectory()', () => {
         const basePath = path.join('this', 'is', 'the', 'way')
         const extendedPath = path.join(basePath, 'forward')
         const filename = 'yadayadayada.log'
 
-        it('returns true for the same dir', () => {
-            assert.ok(isContainedWithinDirectory(basePath, basePath))
-        })
-
-        it('returns true for a subdir', () => {
-            assert.ok(isContainedWithinDirectory(basePath, extendedPath))
-        })
-
-        it('returns true for a file in the same dir', () => {
-            assert.ok(isContainedWithinDirectory(basePath, path.join(basePath, filename)))
-        })
-
-        it('returns true for a file in a subdir', () => {
-            assert.ok(isContainedWithinDirectory(basePath, path.join(extendedPath, filename)))
-        })
-
-        it('returns false for a completely different dir', () => {
-            assert.ok(!isContainedWithinDirectory(basePath, path.join('what', 'are', 'you', 'looking', 'at')))
-        })
-
-        it('returns false for a similarly named dir', () => {
-            assert.ok(!isContainedWithinDirectory(basePath, `${basePath}point`))
-        })
+        assert.ok(isInDirectory(basePath, basePath))
+        assert.ok(isInDirectory(basePath, extendedPath))
+        assert.ok(isInDirectory(basePath, path.join(basePath, filename)))
+        assert.ok(isInDirectory(basePath, path.join(extendedPath, filename)))
+        assert.ok(!isInDirectory(basePath, path.join('what', 'are', 'you', 'looking', 'at')))
+        assert.ok(!isInDirectory(basePath, `${basePath}point`))
```

Comment:
this is a much more compact way to write these tests without any loss of clarity. The old test granular descriptions aren't helpful: they struggle to find a purpose, and that's a clue that they aren't useful.

Reply:
The value the former approach takes is that you have a full (and in theory, descriptive) list of which scenarios break down when there is a failure. The changed approach fails on first issue, without any high level description (think CI reports) as to what situation is broken, and which scenarios are broken.

A parameterized test is the best fit, a crude version of which can be seen in some places in the testing code.

Reply:
> you have a full (and in theory, descriptive) list of which scenarios break down when there is a failure.

But failure is the uncommon case, and the gain is very small: if this test fails in the new form, it is easy to see the problem. If the next case fails, well, that's a _very uncommon_ scenario, and in any case the cost is an extra test run. That cost vs having verbose, low-density test cases is a questionable tradeoff.

We should optimize for the common case.

> A parameterized test is the best fit, a crude version of which can be seen in some places in the testing code.

Those tests tend to be even harder to read, because the codepath is not lexical. These literal cases are obvious and easy to read.

Reply:
> But failure is the uncommon case, and the gain is very small: if this test fails in the new form, it is easy to see the problem.

I am not comfortable with the spirit of that statement (for any tests, not as it applies to this method being tested). When edge cases reveal a failure, it isn't necessarily easy to see the problem, or which scenarios are affected. You only know that _something_ is wrong with foo(). Getting a result that says "expected: true, actual: false" doesn't get you any closer to knowing where the issue is, which requires more effort to triage.

Reply:
It is not a general principle, it is context-sensitive. 

In fact most of the suggestions I find myself advocating fall under the general idea that we should use more discerning practices.  We should not be hyper-granular just out of habit. I'm 100% in favor of descriptive test cases for more complicated functions. `isInDirectory()` is an isolated piece of logic, which may take infinite inputs. Having infinite test cases to model those inputs makes tests harder to read and harder to write, and optimizes for a very uncommon scenario.